platforms/documentation/docs/src/docs/userguide/dep-man/dependency-management/dependency_locking.adoc

```diff
@@ -0,0 +1,301 @@
+====
+
+Locking of a configuration happens through the link:{groovyDslPath}/org.gradle.api.artifacts.ResolutionStrategy.html[ResolutionStrategy] API:
+
+====
+include::sample[dir="snippets/dependencyManagement/dependencyLocking-lockingSingleFilePerProject/kotlin",files="build.gradle.kts[tags=locking-single]"]
+include::sample[dir="snippets/dependencyManagement/dependencyLocking-lockingSingleFilePerProject/groovy",files="build.gradle[tags=locking-single]"]
+====
+
+Only configurations that can be resolved will have lock state attached to them.
+Applying locking on non resolvable-configurations is a no-op.
+
+== Locking all configurations
+
+The following locks all configurations:
+
+====
+include::sample[dir="snippets/dependencyManagement/dependencyLocking-lockingAllConfigurations/kotlin",files="build.gradle.kts[tags=locking-all]"]
+include::sample[dir="snippets/dependencyManagement/dependencyLocking-lockingAllConfigurations/groovy",files="build.gradle[tags=locking-all]"]
+====
+
+The above will lock all _project_ configurations, but not the _buildscript_ ones.
+
+== Disable locking specific configurations
+
+You can also disable locking on a specific configuration.
+
+This can be useful if a plugin configured locking on all configurations, but you happen to add one that should not be locked:
+
+====
+include::sample[dir="snippets/dependencyManagement/dependencyLocking-unlockingSingleConfiguration/kotlin",files="build.gradle.kts[tags=locking-one]"]
+include::sample[dir="snippets/dependencyManagement/dependencyLocking-unlockingSingleConfiguration/groovy",files="build.gradle[tags=locking-one]"]
+====
+
+== Locking buildscript classpath configuration
+
+If you apply plugins to your build, you may want to leverage dependency locking there as well.
+
+To lock the <<plugins.adoc#sec:applying_plugins_buildscript,`classpath` configuration>> used for script plugins:
+
+====
+include::sample[dir="snippets/dependencyManagement/dependencyLocking-lockingClasspathConfiguration/kotlin",files="build.gradle.kts[tags=locking-classpath]"]
+include::sample[dir="snippets/dependencyManagement/dependencyLocking-lockingClasspathConfiguration/groovy",files="build.gradle[tags=locking-classpath]"]
+====
+
+== Generating and updating dependency locks
+
+To generate or update the lock state, use the `--write-locks` command line argument along with the tasks that would trigger configurations to be resolved:
+
+[source,text]
+----
+$ ./gradlew dependencies --write-locks
+----
+
+This will create or update the lock state for each resolved configuration during that build execution.
```

Comment:
💅 I know you cover it in the next section, but I'm immediately wondering what this lock state is.  Perhaps in the intro section you could give a very brief overview of the process, and that the lock state is saved to a file.

Reply:
Added an intro paragraph talking about lock state and files.